### PR TITLE
Auto populated select menus

### PR DESCRIPTION
**Pull request by FedorLap2006**
Implements https://github.com/discord/discord-api-docs/pull/5543, auto populated (with users, channels and roles) select menus, in `SelectMenu` data structure.

### CODE REVIEW

**Comment by Nv7-GitHub** [comment]
Yay its out of draft

**Comment by FedorLap2006** [comment]
Have a few things to finish still

File: examples/components/main.go [hunk 1]
```diff
@@ -438,6 +474,10 @@ func main() {
 		log.Fatalf("Cannot create slash command: %v", err)
 	}
 
+	if err != nil {
+		log.Fatalf("Cannot create slash command: %v", err)
+	}
+
```
Comment (FedorLap2006) [inline examples/components/main.go:480]:
```suggestion
```

File: components.go [hunk 2]
```diff
@@ -169,8 +174,23 @@ type SelectMenuOption struct {
 	Default bool `json:"default"`
 }
 
+// SelectMenuType represents select menu types.
```
Comment (FedorLap2006) [inline components.go:177]:
```suggestion
// SelectMenuType represents select menu type.
```

**Comment by FedorLap2006** [comment]
Finally... it's done

**Comment by Nv7-GitHub** [comment]
yay

**Pull request merged**